samples/matter/weather_station/src/app_task.cpp

```diff
@@ -184,18 +182,10 @@ void AppTask::ButtonStateHandler(uint32_t buttonState, uint32_t hasChanged)
 
 void AppTask::FunctionTimerHandler(AppEvent *)
 {
-	switch (sFunctionTimerMode) {
-	case FunctionTimerMode::kFactoryResetTrigger:
-		LOG_INF("Factory Reset triggered. Release button within %ums to cancel.",
-			kFactoryResetCompleteTimeoutMs);
-		sFunctionTimerMode = FunctionTimerMode::kFactoryResetComplete;
-		k_timer_start(&sFunctionTimer, K_MSEC(kFactoryResetCompleteTimeoutMs), K_NO_WAIT);
-		break;
-	case FunctionTimerMode::kFactoryResetComplete:
+	if (sIsTimerEnabled) {
```

Comment:
Why is the variable needed? If this function gets called the surely the timer is enabled, no?